turbo/jsonrpc/erigon_receipts.go

```diff
@@ -301,9 +322,10 @@ func (api *ErigonImpl) GetLatestLogs(ctx context.Context, crit filters.FilterCri
 				blockLogs = append(blockLogs, filtered[i])
 				logCount++
 			}
-			if logOptions.LogCount != 0 && logOptions.LogCount == logCount {
-				continue
+			if logOptions.LogCount != 0 && logOptions.LogCount <= logCount {
+				break
```

Comment:
why break only internal loop? maybe need break both?

Reply:
here is 2 loops:
```
for iter.HasNext() {
it, err := tx.Prefix(kv.Log, hexutility.EncodeTs(blockNumber))
		if err != nil {
			return nil, err
		}
		for it.HasNext() {
                       // your break is here, and will break only internal loop
```

Reply:
here is 2 loops:

for iter.HasNext() {
                it, err := tx.Prefix(kv.Log, hexutility.EncodeTs(blockNumber))
		if err != nil {
			return nil, err
		}
		for it.HasNext() {
                       fill blockLogs list
                       if logOptions.LogCount != 0 && logOptions.LogCount <= logCount {
                                break
                 }
                 
                 fills erigonLog from data of blockLogs and others
                 
                  if logOptions.LogCount != 0 && logOptions.LogCount <= logCount {
                        return erigonLogs, nil
                }
                if logOptions.BlockCount != 0 && logOptions.BlockCount <= blockCount {
                        return erigonLogs, nil
                }

      }

Reply:
Ok